packages/app/src/components/DimensionsPanel/Dialogs/DialogManager.js

```diff
@@ -322,23 +319,44 @@ export class DialogManager extends Component {
 
     render() {
         const { dialogId, dimensions } = this.props
-        const keepMounted = !dialogId || dialogId === DIMENSION_ID_ORGUNIT
+
+        const nonPredefinedDimensions = filterOutPredefinedDimensions(
+            Object.keys(this.props.dimensions)
+        )
+
+        let dialogTitle = ''
+        if (dialogId === DIMENSION_ID_DATA) {
+            dialogTitle = i18n.t('Data')
+        } else if (dialogId === DIMENSION_ID_PERIOD) {
+            dialogTitle = i18n.t('Period')
+        } else if (dialogId === DIMENSION_ID_ORGUNIT) {
+            dialogTitle = i18n.t('Organisation units')
+        } else if (nonPredefinedDimensions.includes(dialogId)) {
+            dialogTitle = dimensions[dialogId] && dimensions[dialogId].name
+        }
 
         return (
-            <Dialog
-                data-test="dialog-manager"
-                open={dialogId in dimensions}
-                onClose={this.closeDialog}
-                maxWidth="lg"
-                disableEnforceFocus
-                keepMounted={keepMounted}
-            >
-                {this.renderDialogContent()}
-                <DialogActions>
-                    <HideButton onClick={this.closeDialog} />
-                    {dialogId && this.renderPrimaryButton(dialogId)}
-                </DialogActions>
-            </Dialog>
+            <Fragment>
+                {dialogId in dimensions && (
+                    <Modal
+                        onClose={this.closeDialog}
+                        data-test="dialog-manager"
+                        position="top"
+                        large
+                    >
+                        <ModalTitle>{dialogTitle}</ModalTitle>
+                        <ModalContent>
+                            {this.renderDialogContent()}
+                        </ModalContent>
+                        <ModalActions>
+                            <ButtonStrip>
+                                <HideButton onClick={this.closeDialog} />
+                                {dialogId && this.renderPrimaryButton(dialogId)}
```

Comment:
Is this `dialogId` check necessary?
It should not happen that `dialogId` is `undefined` at this point...

Reply:
Good catch, just legacy stuff I suppose. Removed it and consolidated the `renderPrimaryButton` fn in to the code as well.